addons/website_crm_partner_assign/static/src/js/crm_partner_assign.js

```diff
@@ -1,202 +1,241 @@
 odoo.define('crm.partner_assign', function (require) {
 'use strict';
 
+var sAnimations = require('website.content.snippets.animation');
 var rpc = require('web.rpc');
 var weContext = require('web_editor.context');
-require('web.dom_ready');
 /*
  * This file is intended to add interactivity to survey forms rendered by
  * the website engine.
  */
 
-var interested_form = $('.interested_partner_assign_form');
-var desinterested_form = $('.desinterested_partner_assign_form');
-var opp_stage_buttons = $('.opp-stage-button');
-var new_opp_form = $('.new_opp_form');
+sAnimations.registry.crmPartnerAssign = sAnimations.Class.extend({
+    selector: "#portal_my_lead:has(.interested_partner_assign_form,.desinterested_partner_assign_form,.opp-stage-button,.new_opp_form')",
+    read_events: {
+        'click .interested_partner_assign_confirm': '_onInterestedPartnerAssignConfirm',
+        'click .desinterested_partner_assign_confirm': '_onDesinterestedPartnerAssignConfirm',
+        'click .opp-stage-button': '_onOppStageButtons',
+        'change .edit_contact_form .country_id': '_onEditContactForm',
+        'click .edit_contact_confirm': '_onEditContactConfirm',
+        'click .new_opp_confirm': '_onNewOppConfirm',
+        'click .edit_opp_confirm': '_onEditOppConfirm',
+        'change .edit_opp_form .next_activity': '_onEditOppConfirm',
+        'click div.input-group span.fa-calendar': '_onCalendarIcon'
+    },
 
-if(!interested_form.length && !desinterested_form.length && !opp_stage_buttons.length && !new_opp_form.length) {
-    return $.Deferred().reject("DOM doesn't contain website_crm_partner_assign elements");
-}
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
 
-$('.interested_partner_assign_confirm').on('click',function(e){
-    var $btn = $(this);
-    if ($('.interested_partner_assign_form .comment_interested').val() && $('.interested_partner_assign_form .contacted_interested').prop('checked')){
+    /**
+     * @override
+     * @param {Object} ev
+     */
+    _onInterestedPartnerAssignConfirm: function (ev) {
+        var $btn = $(ev.currentTarget);
+        if ($('.interested_partner_assign_form .comment_interested').val() && $('.interested_partner_assign_form .contacted_interested').prop('checked')){
+            $btn.prop('disabled', true);
+            this._rpc({
+                    model: 'crm.lead',
+                    method: "partner_interested",
+                    args: [
+                        [parseInt($('.interested_partner_assign_form .assign_lead_id').val())],
+                        $('.interested_partner_assign_form .comment_interested').val()
+                    ],
+                    context: weContext.get(),
```

Comment:
You can remove the context if it is `weContext.get()` as it is automatic with `this._rpc`. You may then be able to remove the weContext requirement.

Reply:
@qsm-odoo what about rpc.query ? is it the same as this._rpc ? :p

Reply:
`rpc.query` can (must) be replaced by `this._rpc` when possible (when inside a Widget / Animation). And only with `this._rpc`: you can remove the context if equal to `weContext.get()` as it is the default when using `this._rpc`.

In the end both do the same thing but `rpc.query` or even `ajax.jsonRpc` is the old way. The new `this._rpc` allows to avoid to set the context by hand and can also be intercepted by a parent (example: for our JS tests all rpc are intercepted that way to simulate the server response; we do not have JS tests for website yet but when we will, `this._rpc` will be required).